debug/gdbserver.py

```diff
@@ -1649,14 +1649,16 @@ def setup(self):
         # pylint: disable=attribute-defined-outside-init
         self.gdb.load()
 
-        misa = self.hart.misa
+        # It may be affected by reset halt
+        misa = self.gdb.p(f"$misa=0x{self.hart.misa:x}")
         self.supported = set()
         if misa & (1<<20):
             self.supported.add(0)
         if misa & (1<<18):
             self.supported.add(1)
         if misa & (1<<7):
-            self.supported.add(2)
+            self.supported.add(4)
+            self.supported.add(5)
```

Comment:
Moreover, AFAIU, `H` extension does not require presence of both `S` and `U` extensions.
Therefore, I'd suggest something like:
```suggestion
            self.supported_vmodes = self.supported
        else:
            self.supported_vmodes = None
```

Reply:
Thanks, addressed